staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -142,20 +142,32 @@ type ResourceVersionUpdater interface {
 // should be offloaded.
 type WatchErrorHandler func(r *Reflector, err error)
 
-// DefaultWatchErrorHandler is the default implementation of WatchErrorHandler
-func DefaultWatchErrorHandler(r *Reflector, err error) {
+// The WatchErrorHandler is called whenever ListAndWatch drops the
+// connection with an error. After calling this handler, the informer
+// will backoff and retry.
+//
+// The default implementation looks at the error type and tries to log
+// the error message at an appropriate level.
+//
+// Implementations of this handler may display the error message in other
+// ways. Implementations should return quickly - any expensive processing
+// should be offloaded.
+type WatchErrorHandlerWithContext func(ctx context.Context, r *Reflector, err error)
+
+// DefaultWatchErrorHandler is the default implementation of WatchErrorHandlerWithContext.
+func DefaultWatchErrorHandler(ctx context.Context, r *Reflector, err error) {
```

Comment:
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126387/pull-kubernetes-apidiff/1818538357228048384:
```
Incompatible changes:
- ./tools/cache.DefaultWatchErrorHandler: changed from func(*Reflector, error) to func(context.Context, *Reflector, error)
```

https://grep.app/search?q=DefaultWatchErrorHandler

Some copies seem to exist, but not used elsewhere?

Users of `cache.Config` can still set a watch handler without the extra context parameter.

Reply:
both opa gatekeeper and kubevirt aren't libraries. So version skew shouldn't be an issue.